### PR TITLE
Add register process

### PR DESCRIPTION
**Pull request by KhaledBousrih**

### CODE REVIEW

File: concrete_datastore/api/v1_1/serializers.py [hunk 1]
```diff
@@ -29,6 +29,12 @@
 concrete = apps.get_app_config('concrete')
 
 
+class ProcessRegisterSerializer(serializers.Serializer):
+    application = serializers.CharField(max_length=200, required=True)
+    instance = serializers.CharField(max_length=200, required=True)
+    token = serializers.CharField(max_length=40, required=True)
```
Comment (lcognat) [inline concrete_datastore/api/v1_1/serializers.py]: Why such a limitation ? If an uuid v4 is not required (Char here), a 255 char long should be possible.

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 ### Added
 
-- nothing added
+- add process autoregister
```
Comment (lcognat) [inline CHANGELOG.md]: Description of feature added is not long enough

File: concrete_datastore/api/v1_1/urls.py [hunk 3]
```diff
@@ -113,6 +114,11 @@
         GenerateSecureTokenApiView.as_view(),
         name='generate-secure-token',
     ),
+    re_path(
+        r'process/register/',
+        ProcessRegisterApiView.as_view(),
+        name='process-register',
```
Comment (lcognat) [inline concrete_datastore/api/v1_1/urls.py]: the name of the view is not explicit enough. 

The view above is named "generate ... token" as it is used to generate a token. This view does not "process" anything. Maybe "register-as-process"

File: CHANGELOG.md (hunk 2 above)
Comment (KhaledBousrih) [reply CHANGELOG.md]: more details are added

File: concrete_datastore/api/v1_1/urls.py (hunk 3 above)
Comment (KhaledBousrih) [reply concrete_datastore/api/v1_1/urls.py]: name changed to `register-as-process `

File: concrete_datastore/api/v1_1/serializers.py (hunk 1 above)
Comment (KhaledBousrih) [reply concrete_datastore/api/v1_1/serializers.py]: this token is used to generate un instance of `AuthToken` with `key=token`. AuthToken enherits from `from rest_framework.authtoken.models.Token` (the default `key` field has a max_length of 40 chars). So if we increase `max_length` of the serialiser's field `token` we shuld also increase it in field `key` of `AuthToken`. What do you think ?

Comment (lcognat) [reply concrete_datastore/api/v1_1/serializers.py]: In that case, the serializer field should be an instance of `serializers.UUIDField`

**Pull request merged**